firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigContainer.java

```diff
@@ -86,11 +86,18 @@ private ConfigContainer(
    * <p>The {@code containerJson} must not be modified.
    */
   static ConfigContainer copyOf(JSONObject containerJson) throws JSONException {
+    // Personalization metadata may not have been written yet.
+    JSONObject personalizationMetadataJSON =
+        containerJson.optJSONObject(PERSONALIZATION_METADATA_KEY);
```

Comment:
Thinking out loud: `getJSONObject` throws if the given key isn't found, which would result in the new `ConfigContainer` not being created, as noted in https://github.com/firebase/firebase-android-sdk/issues/2186#issuecomment-736355842 Metadata depends on integrations with RC and may not be defined, ie it's not necessarily an error case, and may be a common case (eg for a project that never uses metadata). Using `optJSONObject` instead facilitates checking for null and assigning a default value rather than throwing.